test/python/qasm3/test_export.py

```diff
@@ -2665,6 +2665,24 @@ def test_switch_v1_expr_target(self):
         test = dumps(qc, experimental=ExperimentalFeatures.SWITCH_CASE_V1)
         self.assertEqual(test, expected)
 
+    def test_circuit_with_unitary(self):
+        """Test that circuits with `unitary` gate are correctly handled"""
+        matrix = [[0, 1], [1, 0]]
+        qc = QuantumCircuit(1)
+        qc.unitary(matrix, [0])
+        expected = """\
+OPENQASM 3.0;
+include "stdgates.inc";
+gate unitary _gate_q_0 {
+  U(pi, -pi, 0) _gate_q_0;
+}
+qubit[1] q;
+unitary q[0];
+"""
+        test = dumps(qc)
+        print(test)
```

Comment:
```suggestion
```

If the test fails, the assertion should print out both strings - it's the job of the test runner to display those nicely (if it was a problem for you, likely we can tweak some settings somewhere).

Reply:
This one's probably a debug print I forgot to remove - my bad.

Reply:
Oh yeah, sure, my thing is that the test runner should show the exact same string on an assertion failure, so the debug print shouldn't be necessary.  If it's still necessary, it probably indicates that we need better configuration exposed for the test runner or something.